packages/x-charts-pro/src/Heatmap/Heatmap.tsx

```diff
@@ -100,6 +101,20 @@ const defaultColorMap = interpolateRgbBasis([
 
 const seriesConfig: ChartSeriesConfig<'heatmap'> = { heatmap: heatmapSeriesConfig };
 
+function getDefaultDataForAxis(series: HeatmapProps['series'], dimension: number) {
+  if (series?.[0]?.data === undefined || series[0].data.length === 0) {
+    return [];
+  }
+
+  const uniqueValues = new Set();
+
+  series[0].data.forEach((dataPoint) => uniqueValues.add(dataPoint[dimension]));
+
+  return Array.from(uniqueValues.values());
```

Comment:
You just need the maximal value in the direction, because the 2 first numbers in heatmap series.data are respectively the x and y indexes of the cell.

So default values should go from 0 to the maximal index

It's your test that let me realised that. The `[-1, 3, 7]` can not be displayed because `-1` is not be a valid index
![image](https://github.com/user-attachments/assets/ea03bea7-8e8a-45fa-82e5-bde5d1b4bbb8)

Reply:
This heatmap API seems a bit weird to me. If the x- and y-axes are band scales, why can't I set any value I want? 

Nevertheless, I'll update this to your proposed solution 👍

Reply:
> If the x- and y-axes are band scales, why can't I set any value I want?

That's the purpose, you can put any value in the x/y (dates, numbers, string). The series is independent from them since it relies only on indexes. Avoiding types like
```
seris.data: [(Date | number | string), (Date | number | string), number]
```

Technically, you could put any thing as a default value in `data` Just you need to have enough items to display the min/max indexes